src/NHSD.BuyingCatalogue.Solutions.Persistence/Repositories/SolutionRepository.cs

```diff
@@ -88,8 +92,18 @@ FROM dbo.Solution
         /// <param name="id">The ID of a <see cref="ISolutionResult"/>.</param>
         /// <param name="cancellationToken">A token to notify if the task operation should be cancelled.</param>
         /// <returns>A task representing an operation to retrieve a <see cref="ISolutionResult"/> matching the specified ID.</returns>
-        public async Task<ISolutionResult> ByIdAsync(string id, CancellationToken cancellationToken) =>
-            (await dbConnector.QueryAsync<SolutionResult>(ByIdSql, cancellationToken, new { id })).SingleOrDefault();
+        public async Task<ISolutionResult> ByIdAsync(string id, CancellationToken cancellationToken)
+        {
+            var solutions = (await dbConnector.QueryAsync<SolutionResult>(ByIdSql, cancellationToken, new { id })).ToList();
+            ISolutionResult solution = solutions.FirstOrDefault();
+            if (solutions.Count > 0)
+            {
+                solution.FrameworkShortNames = new List<string>();
+                solution.FrameworkShortNames = solutions.Select(x => x.FrameworkShortName).ToList();
+            }
+
+            return solution;
```

Comment:
This doesn't feel like the right way to implement this. Can you have a look at how `Capabilities` of `SolutionSummaryResult` has been implemented and see if a similar approach can be used here?

There might be an alternative approach to this using SQL too – I might have a look into that.

Reply:
Updated Implementation,

- Added new SolutionFrameworkRepository
- Updated FrameworkSolution JOIN query
- Updated Test cases with FrameworkSolution